smarts/core/provider.py

```diff
@@ -46,3 +48,29 @@ def filter(self, vehicle_ids):
                 del self.vehicles[index]
             except ValueError:
                 continue
+
+
+class Provider:
+    """A Provider tracks a (sub)set of vehicles that all share the same action space.
+    This is a base class (interface) from which all Providers should inherit."""
```

Comment:
Should we make the existing providers inherit from this interface? So from `SumoTrafficSimulation`, `MotionPlannerProvider` and `TrafficHistoryProvider`

Reply:
Yes.  I thought I had!  Hmm... I think I went too far when I backed out the code where `Renderer` inherited it.  I will re-add.